### DISCUSSION TITLE
Required option not respected on product list buy button

### DISCUSSION THREAD

**Discussion by alex-pommier** [openstore-ecommerce/OpenStore#286] [Q&A]
On the product list page there is a "Buy" button that adds the item selected directly into the cart regardless if the option is set to required or not so if the customer checks out having added items in this way the order information will not display the selected required option. I'm not sure if this is a configuration error on my part or a bug?

**Comment by DNNMonster** [comment]
I think you've identified a bit of a problem with the templates.  The ProductDisplayListBody.cshtml file ensures that users visit the details page to select a model when a product has more than one but it should probably be doing the same for products with options. 

I pushed a bugfix branch with an additional check for options that I think is proper. It'll get merged if nobody raises a red flag. You can also easily apply it to a custom template for your site if you need it before the next official release.

**Reply by alex-pommier** [reply]
@DNNMonster Thank you I applied the change it's working as expected now.
